src/lib/Bucket.ts

```diff
@@ -1,3 +1,45 @@
-class Bucket {}
+const wait = (time: number) => new Promise(rs => setTimeout(rs, time));
+class Bucket {
+    #queue: Function[] = [];
```

Comment:
Why did you use a private variable instead of an access modifier? Everything else looks good but this is a bit inconsistent... unless it is necessary?

Reply:
well, not necessary, rather by design

Reply:
You wish for the UX to enforce strict mutability? I can roll with that.

Reply:
I believe using a ES private field for this is a better idea, but not strictly required